src/angular/icons/icons.component.ts

```diff
@@ -0,0 +1,52 @@
+import {Component, Input, PipeTransform, Pipe} from "@angular/core";
+import { DomSanitizer } from '@angular/platform-browser';
+import { Icons } from '../../iconsMap';
+
+export enum Positions{
+    Left,
+    Right,
+    Bottom,
+    Top
+}
+
+export enum Colors{
+    Warning,
+    Secondary,
+    Positive,
+    Negative,
+    Primary
+}
+
+@Pipe({ name: 'safeHtml'})
+
+export class SafeHtmlPipe implements PipeTransform  {
+    constructor(private sanitized: DomSanitizer) {}
+    transform(value) {
+        return this.sanitized.bypassSecurityTrustHtml(value);
+    }
+}
+
+
+@Component({
+    selector:'sdc-icon',
+    templateUrl:'./icons.component.html'
+})
+
+
+export class IconsComponent{
+    @Input() iconName: string;
+    @Input() label : string;
+    @Input() labelPosition : Positions;
+    @Input() color:string;
+    @Input() disabled: boolean;
+
+    public positions = Positions;
+    public colors = Colors;
+
+    public  getFontIconCode(name):string  {
+        let iconhtml;
+        Icons[this.iconName] ? iconhtml =  Icons[this.iconName] : iconhtml ='<span class="svg-icon-missing">Missing Icon</span>';
```

Comment:
```
return Icons[this.iconName] ? Icons[this.iconName] : '<span class="svg-icon-missing">Missing Icon</span>';
```

More over, you can put that in a class attribute, and change it whenever `iconName` is changed on `ngOnChanges`.

Reply:
I think its better to leave it like this - because this function give final result of HTML string which we push into DOM, because of it I think it's good solution to send different lines of HTML based on result.

Reply:
@eliyahen Do you want to make the component able to change icons dynamically? Otherwise I agree that the output should probably be the final HTML